scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,1122 @@
+            if size is None:
+                return out_discr[0]
+            return np.asarray(out_discr).reshape(size)
+        else:
+            raise NotImplementedError("only univariate continuous and "
+                                      "discrete distributions supported")
+
+    @property
+    def seed(self):
+        return self.numpy_rng
+
+    @seed.setter
+    def seed(self, new_seed):
+        self.numpy_rng = get_numpy_rng(new_seed)
+        _lock.acquire()
+        try:
+            self._messages.clear()
+            unur_set_stream(self._messages.handle)
+            unur_urng_free(self.urng)
+            self._urng_builder = _URNG(self.numpy_rng)
+            self.urng = self._urng_builder.get_urng()
+            if self.urng == NULL:
+                raise UNURANError(self._messages.get())
+            unur_chg_urng(self.rng, self.urng)
+        finally:
+            _lock.release()
+
+    @cython.final
+    def __dealloc__(self):
+        if self.distr != NULL:
+            unur_distr_free(self.distr)
+            self.distr = NULL
+        if self.par != NULL:
+            unur_par_free(self.par)
+            self.par = NULL
+        if self.rng != NULL:
+            unur_free(self.rng)
+            self.rng = NULL
+        if self.urng != NULL:
+            unur_urng_free(self.urng)
+            self.urng = NULL
+
+    # Pickling support
+    @cython.final
+    def __reduce__(self):
+        klass = functools.partial(self.__class__, **self._kwargs)
+        return (klass, ())
+
+
+cdef class TransformedDensityRejection(Method):
+    r"""
+    Transformed Density Rejection (TDR) Method.
+
+    TDR is an acceptance/rejection method that uses the concavity of a
+    transformed density to construct hat function and squeezes automatically.
+    Most universal algorithms are very slow compared to algorithms that are
+    specialized to that distribution. Algorithms that are fast have a slow
+    setup and require large tables. The aim of this universal method is to
+    provide an algorithm that is not too slow and needs only a short setup.
+    This method can be applied to univariate and unimodal continuous
+    distributions with T-concave density function. See [1]_ and [2]_ for
+    more details.
+
+    Parameters
+    ----------
+    dist : object
+        An instance of a class with 'pdf' and 'dpdf' methods.
+
+        * 'pdf': PDF of the distribution. The signature of the PDF is
+          expected to be: ``def pdf(x: float, *params) -> float``. i.e.
+          the PDF should accept a Python float and the parameters and
+          return a Python float. It doesn't need to integrate to 1 i.e.
+          the PDF doesn't need to be normalized.
+        * 'dpdf': Derivative of the PDF w.r.t x (i.e. the variate). Must
+          have the same signature as the PDF.
+
+    mode : float, optional
+        Mode of the distribution. It is optional and only required if
+        `use_mode` is ``True``.
```

Comment:
Is it used when `use_mode` is `False`? If not, why do we need two parameters? If the user provides a value, then `use_mode` is `True`.

Reply:
> Is it used when `use_mode` is `False`?

No.

> If not, why do we need two parameters? If the user provides a value, then `use_mode` is `True`.

My bad. Of course, we can just use one of the parameters. Thanks for noticing this!!

Reply:
Done.

Reply:
It would be nice if there were more information about how `mode` is used and how it is used differently than `center`. But I'll consider this resolved.

Reply:
> It would be nice if there were more information about how `mode` is used and how it is used differently than `center`. But I'll consider this resolved.

Even UNU.RAN doesn't do that. I will dig into that papers to see what I can find. As this isn't a blocker, I hope to update this in a follow-up PR.